listener/reconciler/kcp/config_watcher_test.go

```diff
@@ -0,0 +1,240 @@
+	virtualWSManager := &MockVirtualWorkspaceConfigManager{}
+
+	watcher, err := NewConfigWatcher(virtualWSManager, log)
+	require.NoError(t, err)
+
+	// Should not panic or error
+	watcher.OnFileDeleted("/test/config.yaml")
+}
+
+func TestConfigWatcher_LoadAndNotify(t *testing.T) {
+	tests := []struct {
+		name           string
+		configPath     string
+		loadConfigFunc func(configPath string) (*VirtualWorkspacesConfig, error)
+		expectError    bool
+		expectCall     bool
+	}{
+		{
+			name:       "successful_load_and_notify",
+			configPath: "/test/config.yaml",
+			loadConfigFunc: func(configPath string) (*VirtualWorkspacesConfig, error) {
+				return &VirtualWorkspacesConfig{
+					VirtualWorkspaces: []VirtualWorkspace{
+						{Name: "ws1", URL: "https://example.com"},
+						{Name: "ws2", URL: "https://example.org"},
+					},
+				}, nil
+			},
+			expectError: false,
+			expectCall:  true,
+		},
+		{
+			name:       "failed_config_load",
+			configPath: "/test/config.yaml",
+			loadConfigFunc: func(configPath string) (*VirtualWorkspacesConfig, error) {
+				return nil, errors.New("config load error")
+			},
+			expectError: true,
+			expectCall:  false,
+		},
+		{
+			name:       "no_change_handler",
+			configPath: "/test/config.yaml",
+			loadConfigFunc: func(configPath string) (*VirtualWorkspacesConfig, error) {
+				return &VirtualWorkspacesConfig{}, nil
+			},
+			expectError: false,
+			expectCall:  false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			log := testlogger.New().HideLogOutput().Logger
+			virtualWSManager := &MockVirtualWorkspaceConfigManager{
+				LoadConfigFunc: tt.loadConfigFunc,
+			}
+
+			watcher, err := NewConfigWatcher(virtualWSManager, log)
+			require.NoError(t, err)
+
+			// Track change handler calls
+			var handlerCalled bool
+			var receivedConfig *VirtualWorkspacesConfig
+			if tt.name != "no_change_handler" {
+				changeHandler := func(config *VirtualWorkspacesConfig) {
+					handlerCalled = true
+					receivedConfig = config
+				}
+				watcher.changeHandler = changeHandler
+			}
+
+			err = watcher.loadAndNotify(tt.configPath)
+
+			if tt.expectError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+
+			if tt.expectCall {
+				assert.True(t, handlerCalled)
+				assert.NotNil(t, receivedConfig)
+				if tt.name == "successful_load_and_notify" {
+					assert.Equal(t, 2, len(receivedConfig.VirtualWorkspaces))
+				}
+			} else {
+				assert.False(t, handlerCalled)
+			}
+		})
+	}
+}
+
+func TestConfigWatcher_Watch_EmptyPath(t *testing.T) {
+	log := testlogger.New().HideLogOutput().Logger
+	virtualWSManager := &MockVirtualWorkspaceConfigManager{
+		LoadConfigFunc: func(configPath string) (*VirtualWorkspacesConfig, error) {
+			return &VirtualWorkspacesConfig{}, nil
+		},
+	}
+
+	watcher, err := NewConfigWatcher(virtualWSManager, log)
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
```

Comment:
There is a `t.Context()` now since Go 1.24 that can replace `context.Background()`
It's also usable in test suites with `suite.T().Context()`